File: core/src/main/java/by/itechart/writeoffact/dto/CreateWriteOffActDto.java
package by.itechart.writeoffact.dto;

import by.itechart.writeoffact.enums.WriteOffActType;
import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.List;

@Data
public class CreateWriteOffActDto {
    @NotNull
    private Long creatorId;
    @NotEmpty
    private String responsiblePerson;
    @NotNull
    private WriteOffActType writeOffActType;
    @NotEmpty
    private List<CreateWriteOffActGoodsDto> writeOffActGoodsDtoList;
}

File: web-app/online-warehouse/src/app/write-off-act/dto/create-write-off-act.dto.ts
import {WriteOffActTypeEnum} from "./enum/write-off-act-type.enum";
import {CreateWriteOffActGoodsDto} from "./create-write-off-act-goods.dto";

export class CreateWriteOffActDto {
  constructor(
    public creatorId: number,
    public responsiblePerson: number,
    public writeOffActType: WriteOffActTypeEnum,
    public writeOffActGoodsDtoList: Array<CreateWriteOffActGoodsDto>
  ) {
  }
}
